Sources/Datadog/RUM/RUMMonitor/Scopes/RUMUserActionScope.swift

```diff
@@ -77,41 +80,38 @@ internal class RUMUserActionScope: RUMScope, RUMContextProvider {
 
     // MARK: - RUMScope
 
-    func process(command: RUMCommand) -> Bool {
-        if let expirationTime = possibleExpirationTime(currentTime: command.time),
-           allResourcesCompletedLoading() {
-            sendActionEvent(completionTime: expirationTime)
-            return false
+    func process(command: RUMCommand) -> RUMScopeState {
+        if let state = command.apply(to: self) {
```

Comment:
`RUMCommand` is still a concret base class and this will indeed call the implementation of `RUMCommand` here.
I guess it would look a little less odd if it was not called `apply(to:)` but rather `RUMCommand.checkExpiration(with:)`.

Reply:
Agree, some renaming would be handy here. Maybe something like `RUMCommand.getExpirationState(for: self)` would make it more readable in this `if let` context, but I'm okay with any alternative using "expiration" word 👌.